src/Nethermind/Nethermind.Merge.Plugin/Handlers/V1/NewPayloadV1Handler.cs

```diff
@@ -273,64 +264,65 @@ private bool ShouldProcessBlock(Block block, BlockHeader parent, out ProcessingO
             return parentProcessed || processTerminalBlock;
         }
 
-        private async Task<(ValidationResult, string? Message)> ValidateBlockAndProcess(Block block, BlockHeader parent, ProcessingOptions processingOptions)
+        private async Task<(ValidationResult, string?)> ValidateBlockAndProcess(Block block, BlockHeader parent, ProcessingOptions processingOptions)
         {
-            ValidationResult ToValid(bool valid) => valid ? ValidationResult.Valid : ValidationResult.Invalid;
-            string? validationMessage = null;
+            ValidationResult CacheResult(ValidationResult result)
+            {
+                // notice that it is not correct to add information to the cache
+                // if we return SYNCING for example, and don't know yet whether
+                // the block is valid or invalid because we haven't processed it yet
+                if (result != ValidationResult.Syncing)
+                    _latestBlocks?.Set(block.GetOrCalculateHash(), result == ValidationResult.Valid);
+                return result;
```

Comment:
Let's whitelist Valid/Invalid here instead of blacklisting Syncing.
With DB layout changes we can start use Accepted and this can be easily omitted.